File: src/types/specialAbility/sub/_LanguageScript.ts
import * as DB from "tsondb/schema/dsl"
import { ContinentIdentifier } from "../../_Identifier.js"

export const AssociatedContinent = DB.TypeAlias(import.meta.url, {
  name: "AssociatedContinent",
  type: () =>
    DB.Object({
      id: DB.Required({
        comment: "The continent’s identifier.",
        type: ContinentIdentifier(),
      }),
      is_extinct: DB.Required({
        comment: "Is the language considered virtually extinct in this continent?",
        type: DB.Boolean(),
      }),
    }),
})

File: Sources/OptolithDatabaseSchema/GeneratedTypes/specialAbility/sub/_LanguageScript.swift
//
//  _LanguageScript.swift
//  OptolithDatabaseSchema
//

import DiscriminatedEnum

public struct AssociatedContinent: EntitySubtype {
    /// The continent's identifier.
    public let id: ContinentIdentifier
    
    /// Is the language considered virtually extinct in this continent?
    public let isExtinct: Bool

    public init(id: ContinentIdentifier, isExtinct: Bool) {
        self.id = id
        self.isExtinct = isExtinct
    }    
    
    private enum CodingKeys: String, CodingKey {
        case id = "id"
        case isExtinct = "is_extinct"
    }
}
